core/src/ibc/channel_04/manager.rs

```diff
@@ -203,14 +208,15 @@ impl<'a> Manager<'a> {
         }
 
         let client_identifier = self.check_connection_opened(&connection)?;
+        let connection_end = self.query_connection(&connection)?;
 
         let expected = ChannelEnd {
             state: ChannelState::INIT,
             ordering: order,
             counterparty_port_identifier: DEFAULT_PORT.to_string(),
             counterparty_channel_identifier: channel_identifier.clone(),
             // Note: the array should be reversed in the future where `connection` becomes an array.
-            connection_hops: vec![connection.clone()],
+            connection_hops: vec![connection_end.counterparty_connection_identifier],
```

Comment:
Note that ICS javascript example uses just `connection` (given as an argument) here. It's ok not to change if you think this is not important.

Reply:
Since the word "connection" was used in the channel/manager.rs code for the connection identifier, I used the term "connection_end".

We need a consistent naming rule. What do you prefer? Shall we use `connection` for the identifier? Shall we use `connection` for the ConnectionEnd?

Reply:
Ah, I was misunderstanding something. The name `connection` is already used in the function parameter, which represents `connectionHops` in javascript. And I misunderstood that we use `previous` here. 
About your point, yes you're right but I think we can just stick to our convention (`connection` for id and `connection_end` for struct).